sunpy/io/tests/test_fits.py

```diff
@@ -182,3 +181,27 @@ def test_read_memmap():
 
     data, _ = _fits.read(TEST_AIA_IMAGE, memmap=False)[0]
     assert data.base is None
+
+
+def test_merge_multiple_comment_history_entries():
+    header = fits.Header()
+    history = "First history entry\nSecond history entry"
+    comment = "First comment\nSecond comment"
+    for hist in history.split("\n"):
+        header.add_history(hist)
+    for com in comment.split("\n"):
+        header.add_comment(com)
+    file_header = _fits.format_comments_and_history(header)
+    assert file_header["HISTORY"] == history
+    assert file_header["COMMENT"] == comment
```

Comment:
I think this test is 👍 , but to mimic a bit closer what a user is likely to do, a test that does the following would be good:
1. loads a map
2. adds some new history and comment
3. saves the map to FITS
4. Loads it back and checks that the new history/comment roundtrips fine

Reply:
I agree on the need for a roundtrip test that more accurately reflects how this code would actually be used. However, I disagree with step 2. This PR does not add functionality for adding hisotry and comment cards (not that someone couldn't by just adding to the respective entries in meta).  The main point of this PR is to 1) parse comment/history entries such that individual cards are separated by a newline when joining them as a single string and 2) to correctly write these back out to individual cards when writing the metadata dictionary back to a FITS header.

Reply:
👍 - I was kind of guessing at the steps